integrations/google/gmail/events.go

```diff
@@ -0,0 +1,42 @@
+package gmail
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"go.autokitteh.dev/autokitteh/integrations/internal/extrazap"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+func ConstructEvent(ctx context.Context, vars sdkservices.Vars, gmailEvent map[string]any, cids []sdktypes.ConnectionID) (sdktypes.Event, error) {
+	l := extrazap.ExtractLoggerFromContext(ctx)
+
+	// TODO(ENG-1235): Enrich the event with relevant data, using API calls.
+
+	// Convert the raw data to an AutoKitteh event.
+	wrapped, err := sdktypes.DefaultValueWrapper.Wrap(gmailEvent)
```

Comment:
sdktypes.WrapValue

Reply:
Doing this across the repo in a separate PR